### PR TITLE
[Sol->Yul] Bit operations for fixed bytes types.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: libsolidity/codegen/ir/IRGeneratorForStatements.cpp [hunk 1]
```diff
@@ -2557,10 +2557,25 @@ string IRGeneratorForStatements::binaryOperation(
...
-	if (IntegerType const* type = dynamic_cast<IntegerType const*>(&_type))
+	string fun;
+	if (TokenTraits::isBitOp(_operator))
 	{
-		string fun;
-		// TODO: Implement all operations for signed and unsigned types.
+		solAssert(
+			_type.category() == Type::Category::Integer ||
+			_type.category() == Type::Category::FixedBytes,
+		"");
```
Comment (leonardoalt) [inline libsolidity/codegen/ir/IRGeneratorForStatements.cpp:2566]: Is indentation correct here?

File: libsolidity/codegen/ir/IRGeneratorForStatements.cpp [hunk 2]
```diff
@@ -2557,10 +2557,25 @@ string IRGeneratorForStatements::binaryOperation(
...
+		switch (_operator)
+		{
+			case Token::BitOr: fun = "or"; break;
```
Comment (leonardoalt) [inline libsolidity/codegen/ir/IRGeneratorForStatements.cpp]: Hm I thought the style was
```
switch (...)
case ...:
case ...:
```
But now I see the tab also in some code below

**Review by leonardoalt — DISMISSED** [review]
Just some minor style comments, not sure they need to be addressed

**Review by Marenz — DISMISSED** [review]

File: libsolidity/codegen/ir/IRGeneratorForStatements.cpp (hunk 1 above)
Comment (chriseth) [reply libsolidity/codegen/ir/IRGeneratorForStatements.cpp:2566]: Strictly, you could ask for
```
solAssert(
  _type.cat... ||
  _type.cat... FixedBytes
, "");
```
But I think moving the `,` to the previous line is better.

**Comment by chriseth** [comment]
Updated.

**Pull request merged**